### PR TITLE
Way_of_the_cross

### LINKED ISSUE #28: Code Trip "Way of the Cross"

### LINKED ISSUE #46: Code Subtitle support for trips
As user
I want to see colored caption in "Way of the Cross" trip

**Requirements:**

1. Implemented captions with secondary color
2. "Way of the Cross" trip updated with captions

### PR DESCRIPTION
**Pull request by knutm92**
fixes #28
fixes #46 

For now I set the caption color to primaryNormal as I couldn't find the brown one we talked about.

Special character indents for the poematic content didn't work as intended, it would require adding new items to the StepItem class, so I left if as is for now.

### CODE REVIEW

**Comment by darowsluk** [comment]
I get 36 compile errors - will write more info...

File: lib/widgets/tripViewStepSwitch_widget.dart [hunk 1]
```diff
@@ -33,6 +33,18 @@ class TripViewStepSwitch extends StatelessWidget {
...
             children: [
+               Container(
+                alignment: Alignment.topCenter,
+                margin: EdgeInsets.symmetric(horizontal: Constants.insideMargin),
+                child: Text(
+                  tripDataController
+                      .getStepItem(tripDataSelect: tripDataSelect)
+                      .caption,
+                  style: Get.context!.textTheme.bodyText2
+                      ?.copyWith(color: AppColors.primaryNormal),
```
Comment (darowsluk) [inline lib/widgets/tripViewStepSwitch_widget.dart:44]: You can add secondaryNormal color with hex number: 0xFF766257

File: lib/model/tripDatabase.dart [hunk 2]
```diff
@@ -642,86 +642,147 @@ List<TripItem> generateTripItems() {
...
           nextStep: 10,
         ),
         StepItem(
-          imageAsset: "assets/images/trip_03.png",
-          title: "Stacja 11",
+          imageAsset: "assets/images/trip02/trip02_11.jpg",
+          caption: "Buk pospolity w odmianie purpurowej (fagus silvatica purpurea)\nKrwistoczerwone liście symbolizują krew Chrystusa.",
+          title: "Stacja XI\nJezus do Krzyża przybity",
+          description:
+              """„Ludzie potrzebują naszych rąk, by im służyły. Naszych nóg, by ich odwiedzały. Naszych ust, by życzliwie do nich mówiły. Naszych serc, by kochały. Kwiat rozwija się, gdy świeci słońce, a człowiek rozwija się, gdy kocha” (bł. Matka Teresa z Kalkuty).\n\nByś mi, Synu, nisko wisiał, wżdybyś ze mnie pomoc miał;\ngłowę bym Twoją podparła, krew zsiadłą z lica otarła.\nAle Cię nie mogę dosiąc, Tobie, Synu, nic dopomóc.""",
```
Comment (darowsluk) [inline lib/model/tripDatabase.dart:738]: Should be: (św. Matka Teresa z Kalkuty)

File: lib/model/tripDatabase.dart [hunk 3]
```diff
@@ -642,86 +642,147 @@ List<TripItem> generateTripItems() {
...
           nextStep: 13,
         ),
         StepItem(
-          imageAsset: "assets/images/trip_03.png",
-          title: "Stacja 14",
+          imageAsset: "assets/images/trip02/trip02_14.jpg",
+          caption: "Dąb szypułkowy (quercus robur)\nDrzewo to niebawem rozrośnie się potężnie jak dzieło Chrystusa. U Żydów panował zwyczaj chowania zmarłych pod dębami.",
+          title: "Stacja XIV\nJezus złożony do grobu",
+          description:
+              """„Śmierć nie oddziela nas od siebie, ponieważ wszyscy zdążamy tą samą drogą i odnajdziemy się w tym samym miejscu. Nie będziemy nigdy rozłączeni, ponieważ żyjemy dla Chrystusa i teraz jesteśmy złączeni z Chrystusem, idąc ku Niemu. Wszyscy razem będziemy kiedyś z Chrystusem” (św. Szymon z Tesalonik).\n\nMatki, co synaczki macie, ajako się w nich wy kochacie,\nkiedy wam z nich jeden umrze, ciężki ból ma serce wasze;\ncóż ja, com miała jednego, już nie mogę mieć inszego.""",
```
Comment (darowsluk) [inline lib/model/tripDatabase.dart:765]: Check if "ajako" should be separated to "a jako"

File: lib/model/tripDatabase.dart [hunk 4]
```diff
@@ -642,86 +642,147 @@ List<TripItem> generateTripItems() {
...
+          caption: "Drobne sadzonki ostrokrzewu (ilex)\nOstrokrzew symbolizuje ojcostwo, a jako roślina wiecznie zielona -nieśmiertelność.",
+          title: "Stacja XV\nZmartwychwstanie Jezusa",
+          description:
+              """„Potrzebujemy małych i większych nadziei, które dzień po dniu podtrzymują nas w drodze. Jednak bez wielkiej nadziei, która musi przewyższać pozostałe, są one niewystarczające. Tą wielką nadzieją może być jedynie Bóg, który ogarnia wszechświat i który może nam zaproponować i dać to, czego sami nie możemy osiągnąć. Bóg jest fundamentem nadziei, nie jakikolwiek bóg, ale ten Bóg, który ma ludzkie oblicze i umiłował nas aż do końca: każdą jednostkę i ludzkość w całości” (papież Benedykt XVI).\n\nO, niestetyż, miły Panie, toć nie małe rozłączenie;\nprzedtem było miłowanie, a teraz ciężkie wzdychanie.\nCzemuż, Boże Ojcze, nie dbasz, o Synaczka pieczy nie masz?""",
+          previousStep: 13,
+          nextStep: 15,
+        ),
+        StepItem(
+          type: StepType.end,
+          title: "Zakończyłeś drogę krzyżową.",
```
Comment (darowsluk) [inline lib/model/tripDatabase.dart:780]: Perhaps better title would be "Odpust" or "Łaska odpustu" or "Zakończyłeś drogę krzyżową" (without period at the end)

File: lib/model/tripDatabase.dart [hunk 5]
```diff
@@ -642,86 +642,147 @@ List<TripItem> generateTripItems() {
...
+          description:
+              'Do nabożeństwa drogi krzyżowej przywiązana jest łaska odpustu zupełnego, jeśli jesteś w stanie łaski uświęcającej i pomodlisz się w intencjach wyznaczonych przez ojca świetego. Czy jest ktoś, komu chicałbyś ten dar ofiarować?',
```
Comment (darowsluk) [inline lib/model/tripDatabase.dart:782]: Consider: 'Do nabożeństwa drogi krzyżowej przywiązana jest łaska odpustu zupełnego pod zwykłymi warunkami. Odpust można uzyskać dla siebie lub dla jednej osoby zmarłej. "Ukrzyżowany Jezus jest wielkim «odpustem» ofiarowanym przez Ojca ludzkości jako przebaczenie win i zaproszenie do synowskiego życia w Duchu Świętym" (św. Jan Paweł II)'

**Review by darowsluk — CHANGES_REQUESTED** [review]
I am amazed at these short and deep texts of the saints in connection with symbolic plants. May this prayer help many people on their way of the cross.

**Comment by knutm92** [comment]
@darowsluk 
Thank you for the review, I have implemented the suggestions.

p.s. Yes, I agree, this is a very worthwhile text for meditation. I'm really grateful we could use it here.

**Pull request merged**